nevergrad/functions/games/battleship.py

```diff
@@ -0,0 +1,337 @@
+            if failed:  # We failed to cover this x!
+                continue
+
+            if verbose:
+                print("Intermediate board with known ships covered:")
+                print(candidate)
+                print(" ... should cover:")
+                print(self)
+                print("Available ships:", available_ships)
+
+            # Now let us put the remaining ships!
+            while available_ships and not failed:
+                if verbose:
+                    print("Remaining ships :" + str(available_ships))
+                # Pick a ship.
+                s = available_ships[-1]
+                if verbose:
+                    print("We try to put ship " + str(s))
+                # Try to put it!
+                num_failures = 5000
+                failures = 0
+                while failures < num_failures:
+                    try_other_position = False
+                    x = random.randint(0, N-s)
+                    y = random.randint(0, N-1)
+                    dx = 1
+                    dy = 0
+                    if random.choice([False, True]):
+                        tmp = x
+                        x = y
+                        y = tmp
+                        dx = 0
+                        dy = 1
+
+                    for i in range(s):
+                        if candidate.board[x+i*dx][y+i*dy] == 'o' or board[x+i*dx][y+i*dy] == '*':
+                            #print " Arg!"
+                            #print "candidateboard=" + (candidate.board[x+i*dx][y+i*dy]) + " where we try to put our ship. (o impossible!)"
+                            #print "board=" + (board[x+i*dx][y+i*dy]) + " where we try to put our ship. (* impossible)"
+                            try_other_position = True
+                            break
+                    if try_other_position:
+                        #print failures, " failures."
+                        failures += 1
+                        continue
+                    break  # Ok, we have found.
+
+                # If we failed to put this ship we must restart from scratch...
+                if failures >= num_failures:
+                    failed = True
+                    break
+
+                # We can put this ship here.
+                for i in range(s):
+                    candidate.board[x+i*dx][y+i*dy] = 'o'
+                del available_ships[-1]
+
+            if failed:
+                continue
+
+            if verbose:
+                print("All ships installed!")
+                print(candidate)
+
+            # We have a candidate!
+            return candidate
+
+    def boardsize(self):
+        if self.version == 2:
+            return 10
+        return 7
+
+    def sizes(self):
+        if self.version == 2:
+            return [2, 3, 3, 4, 5]
+        return [5, 4]  # [5] + [4]*2 + [3]*7 + [2]*4
+        # return [5, 4, 3, 2, 2] # [5] + [4]*2 + [3]*7 + [2]*4
+
+    def fill(self, s=None):
+        random.seed()
```

Comment:
for reproducibility, please remove this
Also, we've been using numpys random generator in the code so it's better to stick to it instead of the standard library

Reply:
removed
not yet switched to np ==> for later sorry, tight time constraints.